runtime/src/message_processor.rs

```diff
@@ -112,16 +115,19 @@ impl MessageProcessor {
                     Some(&instruction_recorders[instruction_index]);
             }
             let mut time = Measure::start("execute_instruction");
-            let compute_meter_consumption = invoke_context
-                .process_instruction(message, instruction, program_indices, &[], &[])
-                .map_err(|err| TransactionError::InstructionError(instruction_index as u8, err))?;
+            let ProcessInstructionResult {
+                compute_units_consumed,
+                result,
+            } = invoke_context.process_instruction(message, instruction, program_indices, &[], &[]);
```

Comment:
this is a good change, should help to match pre-program-timing to replay-stats. Will be very helpful to investigate TPS drop issues.